### ISSUE TITLE
`rubocop -a` syntax error occurs

### ISSUE THREAD

**Issue by jnst** [bbatsov/rubocop#3475]
`rubocop -a` syntax error occurs in latest version.

``` bash
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```
### sample.rb

create this file.

``` ruby
def func(args)
end

# OK
func({ args: [[{ x: 'hello', y:'world' }]] })

# OK
func({
       args: [[{x: 'hello', y:'world'}]]
     })

# OK
func({
       args: [
         [{x: 'hello', y:'world'}]
       ]
     })

# OK
func({ args: [[
                {x: 'hello', y:'world'}
              ]]})

# NG: syntax error occurs
func({ args: [[{x: 'hello', y:'world'}
              ]]
     })

# NG: syntax error occurs
func({ args: [[
        {x: 'hello', y:'world'}
      ]]
  })

# NG: syntax error occurs
func(
  {args:
     [[
        {x: 'hello', y:'world'}
      ]]
  }
)
```

then run `rubocop -a sample.rb`. (don't need .rubocop.yml)
### result

``` ruby
func(
  args:
     [[
       { x: 'hello', y: 'world' }
     ]]} # <- unneeded curly brace
)
```

**Comment by pascalandermatt** [comment]
Which cop is responsible for this, so I can make a PR

**Comment by bbatsov** [comment]
I can't reproduce this with 0.43. Guess it was fixed.

**Comment by bbatsov** [comment]
Probably `Style/MultilineHashBraceLayout`.

**Comment by pascalandermatt** [comment]
I added some Tests, but I could not fix it:

>   it 'corrects issue #3475 on GitHub' do #Should pass
>     corrected = autocorrect_source(cop, ['func(args: [[',"{ x: 'hello', y: 'world' }",']]})'])
>     expect(corrected).to eq("func(args: [[\n{ x: 'hello', y: 'world' }\n]]))")
>       end

>    it 'corrects issue #3475 on GitHub' do #Should fail GitHub issue 
>         corrected = autocorrect_source(cop, ['func({',"args: [[{x: 'hello', y:'world'}]]",'})'])
>         expect(corrected).to eq("func({\nargs: [[{x: 'hello', y:'world'}]]\n})")
>       end

**Comment by yorikim** [comment]
`BracesAroundHashParameters` creates two actions: removing the left and the right braces. Cop `MultilineHashBraceLayout` replaces action `removing right brace` with `replacing with a closing brace`. 

The shortest case:
```ruby
m({ x: 1
})
```

result:
```ruby
m(x: 1})
```

The simplest solution is adding a field `self.auto_correct_rank` to class `Cop` and execute all cops ordered by this field. Something like:
```ruby
class Cop
   ...

  def self.autocorrect_rank
    100
  end
end

class MultilineHashBraceLayout < Cop
  ...
  
  def self.autocorrect_rank
    90
  end
end

class CopStore < ::Array
  def self.new(*cops)
    super(cops.flatten.sort_by(&:autocorrect_rank))
  end
  
  ...
end
```

**Comment by Drenmi** [comment]
Supplied examples auto-correct properly on current `master`. Closing. 🙂
